### PR TITLE
Support query by "target repo + path"

### PR DESCRIPTION
**Pull request by ruhan1**
Add table for query by 'target store+path'. Insert entries when saving tracking record; add REST endpoint to query.

### CODE REVIEW

File: src/main/java/org/commonjava/service/promote/tracking/cassandra/SchemaUtils.java [hunk 1]
```diff
@@ -35,4 +37,16 @@ public static String getSchemaCreateTableTracking( String keySpace )
                 + "PRIMARY KEY (trackingId, promotionId)"
                 + ");";
     }
+
+    public static String getSchemaCreateTableQueryByPath( String keySpace )
```
Comment (sswguo) [inline src/main/java/org/commonjava/service/promote/tracking/cassandra/SchemaUtils.java:41]: I remembered that all these things are hosed in path-mapped-storage, so should we include this there or keeping here works ?

Comment (ruhan1) [reply src/main/java/org/commonjava/service/promote/tracking/cassandra/SchemaUtils.java:41]: as we spread functions to each service, table schema are with their own service. The promote, tracking, and repo service all have their own table schema within own code.

Comment (sswguo) [reply src/main/java/org/commonjava/service/promote/tracking/cassandra/SchemaUtils.java:41]: yeah, I see. LGTM.

**Review by ligangty — COMMENTED** [review]
LGTM

**Pull request merged**